### ISSUE TITLE
Crash when using Mining patch planner with your mod

### ISSUE THREAD

**Issue by malomaster** [fgardt/factorio-rail-decon-planner#1]
Got this error when tried to use mining patch planner:
```
Mod Rail Deconstruction Planner (1.0.0) got an error.
Error while running event rail-decon-planner::on_player_selected_area (ID 51)
Entity is not rail.
stack traceback:
	[C]: in function 'get_rail_segment_rails'
	__rail-decon-planner__/control.lua:142: in function 'get_rail_segments'
	__rail-decon-planner__/control.lua:243: in function 'mark_segments'
	__rail-decon-planner__/control.lua:333: in function <__rail-decon-planner__/control.lua:330>
```
I tried to mark an ore patch for mining using "Mining patch planner" mod, but somehow got an error in "Rail Deconstruction Planner". I think both mods used the same functions and maybe the same locals (function names or smth), so using one mod also tried to call some function from another mod or something like that.
I'm also duplicated this error to "Mining patch planner" author.

**Comment by fgardt** [comment]
No need to duplicate this, apparently I am not correctly checking for the tool/entity.
I'll take a look at this soon (maybe tomorrow, at least monday).

**Comment by fgardt** [comment]
issue ref: https://github.com/rimbas/mining-patch-planner/issues/7

**Comment by fgardt** [comment]
should be fixed in [v1.0.1](https://github.com/fgardt/factorio-rail-decon-planner/releases/tag/v1.0.1) with cc811dad6621afbfe12c52305d25af0169865d8c
